x/adminmodule/keeper/proposal.go

```diff
@@ -7,22 +7,11 @@ import (
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	govv1types "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
-
-	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govv1types "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 )
 
 // SubmitProposal create new proposal given a content
-func (k Keeper) SubmitProposal(ctx sdk.Context, content govv1types.Content) (govv1types.Proposal, error) {
-	if !k.rtr.HasRoute(content.ProposalRoute()) {
-		return govv1types.Proposal{}, sdkerrors.Wrap(govtypes.ErrNoProposalHandlerExists, content.ProposalRoute())
-	}
-
-	cacheCtx, _ := ctx.CacheContext()
-	handler := k.rtr.GetRoute(content.ProposalRoute())
```

Comment:
why did you remove this cached call? This allowed us to get the error message on unsuccessful handler call

Reply:
ah, have the solution here https://github.com/cosmos/cosmos-sdk/pull/10868#discussion_r784872842

Reply:
@pr0n00gler are we gonna implement the call of handler right away instead of waiting for the Endblocker?